data-conversion/src/main.rs

```diff
@@ -33,10 +39,15 @@ fn main() {
         "float".to_string(),
         "hex".to_string(),
         "fixed".to_string(),
+        "exp".to_string(),
     ];
     // Array of all possible options to convert to
-    let to_types: Vec<String> =
-        vec!["binary".to_string(), "float".to_string(), "hex".to_string()];
+    let to_types: Vec<String> = vec![
+        "binary".to_string(),
+        "float".to_string(),
+        "hex".to_string(),
+        "fixed".to_string(),
+    ];
```

Comment:
Another Argh trick you can use: instead of using strings for these, you can use a Rust enum. You declare a Rust enum like this:

```rs
enum NumType {
  Binary,
  Float,
  Hex,
  Fixed,
}
```

You will then need to implement the `FromStr` and `ToString` traits for this type. Something like this:

```rs
impl ToString for NumType {
  fn to_string(&self) -> String {
    match self {
      NumType::Float => "float".to_string(),
      ...
    }
  }
}
```

This will help you avoid all the boilerplate of manual string detection.